### ISSUE TITLE
`<AutocompleteInput>` should pass TextFieldProps to its rendered TextField

### ISSUE THREAD

**Issue by zikowang** [marmelab/react-admin#8357]
**What you were expecting:**
When I crate a custom theme and pass custom `defaultProps` to `MuiTextField` and/or `RaTextField` (in my situation it is the `autoComplete` prop) the `AutocompleteInput` component should render its `TextField` with the passed custom prop.
When I render the `AutocompleteInput` with custom `TextFieldProps`, it should also render its `TextField` with these passed props.

**What happened instead:**
The `TextField` is still rendered without the passed custom prop. In my situation it renders the `TextField` with `autocomplete="off"` instead of `autocomplete="new-password"`. The Chrome Browser still adds auto complete options.

![image](https://user-images.githubusercontent.com/16488024/200251098-e15f9762-2bbd-4e49-a392-f8fbe524c319.png)

**Steps to reproduce:**
1. go to https://codesandbox.io/s/mystifying-noether-fwj0kk?file=/src/comments/PostReferenceInput.tsx
2. make sure you have some address autofill data in your browser and make sure you have it turned on.
3. got to createComment view (`/#/comments/create`)
4. click in the Post-Field
5. your browser should now show you some suggestions based on the autofill data saved in your browser

**Related code:**

Component
https://codesandbox.io/s/mystifying-noether-fwj0kk?file=/src/comments/PostReferenceInput.tsx

Customized Theme
https://codesandbox.io/s/mystifying-noether-fwj0kk?file=/src/muiTheme.ts

**Environment**

* React-admin version: 4.5.0
* React version: 18
* Browser: Google Chrome

**Comment by fzaninotto** [comment]
Your CodeSandbox shows exactly the contrary of what you describe: the input field does have an `autocomplete="new-password"`. 

![image](https://user-images.githubusercontent.com/99944/200005027-8a20f3d3-c3e5-468c-9fd5-c80aa876bcf6.png)

To me, there is no bug. 

Could you rephrase your problem?

**Comment by zikowang** [comment]
You're on `/#/posts`, please go to `/#/comments/create` described in 3. of "Steps to reproduce".

**Comment by slax57** [comment]
@zikowang Thanks for updating the description. I reproduced your issue.
However, DevTools shows that the `autoComplete` prop is correctly passed to the underlying MUI `TextField:`

![2022-11-30_10-12](https://user-images.githubusercontent.com/14542336/204755602-27bcd1d4-3879-4079-8651-d35de67202b4.png)

Hence I suspect the issue is on MUI's side.
So I'll close this issue and encourage you to open a new issue on their repository instead.
Thanks
